Update `filetype.vim`: Fixed Gnuplot detection

RobbiZ98 opened the pull request:
Gnuplot files with the ".gnuplot" extension are now correctly recognised (added missing asterisk)

RobbiZ98 committed:
Update filetype.vim: Fixed Gnuplot detection

Gnuplot files with the ".gnuplot" extension are now correctly recognised (added missing asterisk)

RobbiZ98 committed:
Update test_filetype.vim

Added test for recognition of `.gnuplot` filename extension

chrisbra commented on src/testdir/test_filetype.vim:
```diff
@@ -283,7 +283,7 @@ def s:GetFilenameChecks(): dict<list<string>>
     gn: ['file.gn', 'file.gni'],
     gnash: ['gnashrc', '.gnashrc', 'gnashpluginrc', '.gnashpluginrc'],
-    gnuplot: ['file.gpi', '.gnuplot'],
+    gnuplot: ['file.gpi', 'file.gnuplot'],
```
does `*` no longer match `.gnuplot` files itself?

RobbiZ98 committed:
Update `testfile.vim`: Corrected test for Gnuplot recognition

re-added test for recognition of .gnuplot config file, in addition to .gnuplot as filename extension.

In accordance with https://github.com/vim/vim/pull/14243#issuecomment-2012869620
